ui/src/admin/components/ChangePassRow.js

```diff
@@ -0,0 +1,94 @@
+import React, {Component, PropTypes} from 'react'
+
+import OnClickOutside from 'shared/components/OnClickOutside'
+import ConfirmButtons from 'src/shared/components/ConfirmButtons'
+
+class ChangePassRow extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      showForm: false,
+    }
+    this.showForm = ::this.showForm
+    this.handleCancel = ::this.handleCancel
+    this.handleKeyPress = ::this.handleKeyPress
+    this.handleEdit = ::this.handleEdit
+    this.handleSubmit = ::this.handleSubmit
+  }
+
+  showForm() {
+    this.setState({showForm: true})
+  }
+
+  handleCancel() {
+    this.setState({showForm: false})
+  }
+
+  handleClickOutside() {
+    this.setState({showForm: false})
+  }
+
+  handleSubmit(user) {
+    this.props.onApply(user)
+    this.setState({showForm: false})
+  }
+
+  handleKeyPress(user) {
```

Comment:
convention has tended to be placing the `(e) =>` into the onClick below, but this is a nice pattern also